docs/marketplace/dev/front-end/table-design/index.md

```diff
@@ -0,0 +1,361 @@
+---
+title: Table design
+description: This article describes the Table Design in the Components Library.
+template: concept-topic-template
+---
+
+This article describes the Table Design in the Components Library.
+
+## Overview
+
+A Table Component is an arrangement of data in rows and columns, or possibly in a more complex structure 
+(with sorting, filtering, pagination, row selections, infinite scrolling, etc.). 
+It is an essential building block of a user interface.
+A basic able Component is `<spy-table [config]="config"></spy-table>` where `config` is:
+
+- `dataSource` - an array of rows to be displayed on the table.
+- `columns` - an array of columns.
+
+```ts
+const config: TableConfig = {
+  dataSource: {
+    // transforms input data via Data Transformer service
+    type: DatasourceType,
+    transform?: DataTransformerConfig,
+  },
+  columns: [
+    { id: 'col1', title: 'Column #1' },
+    { id: 'col2', title: 'Column #2' },
+    { id: 'col3', title: 'Column #3' },
+  ],
+};
+```
+
+## Architecture
+
+![Table Architecture](https://spryker.s3.eu-central-1.amazonaws.com/docs/Marketplace/dev+guides/Front-end/table-architecture.svg)
+
+### Components
```

Comment:
This section is not clear. What are you trying to explain here?

Reply:
This is the main section of table architecture with a diagram and subsections like features, components, etc.